File: app/src/main/java/com/graduation/farmerfriend/control/RegistrationFragment.java
package com.graduation.farmerfriend.control;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.graduation.farmerfriend.R;
import com.graduation.farmerfriend.control.iot_fragments.IotIntroFragment;
import com.graduation.farmerfriend.databinding.FragmentRegistrationBinding;

public class RegistrationFragment extends Fragment {

    private FragmentRegistrationBinding viewBinding;
    private IotIntroFragment iotIntroFragment;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        viewBinding = FragmentRegistrationBinding.inflate(inflater, container, false);

        iotIntroFragment = new IotIntroFragment();

//        viewBinding.registerButton.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View view) {
//
//            }
//        });
        viewBinding.registerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (savedInstanceState == null) {
                    FragmentManager fragmentManager = requireActivity().getSupportFragmentManager();
                    FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
                    fragmentTransaction.replace(R.id.registration_fragment_container, iotIntroFragment);
                    fragmentTransaction.commit();
                }
            }
        });

        return viewBinding.getRoot();
    }

    @Override
    public void onPause() {
        super.onPause();
        Log.i("testFragmentLifeCycle", "RegistrationFragmentOnPause()");
    }

    @Override
    public void onStop() {
        super.onStop();
        Log.i("testFragmentLifeCycle", "RegistrationFragmentOnStop()");
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        Log.i("testFragmentLifeCycle", "RegistrationFragmentOnDestroyView()");
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Log.i("testFragmentLifeCycle", "RegistrationFragmentOnDestroy()");
    }
}

File: app/src/main/java/com/graduation/farmerfriend/registration/RegistrationFragment.kt
package com.graduation.farmerfriend.registration

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.Navigation
import androidx.navigation.fragment.findNavController
import androidx.navigation.navOptions
import com.graduation.farmerfriend.R
import com.graduation.farmerfriend.databinding.FragmentRegistrationBinding

class RegistrationFragment : Fragment() {

    private lateinit var viewBinding : FragmentRegistrationBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        viewBinding = FragmentRegistrationBinding.inflate(inflater, container, false)

        val options = navOptions {
            anim {
                enter = R.anim.slide_in_right
                exit = R.anim.slide_out_left
                popEnter = R.anim.slide_in_left
                popExit = R.anim.slide_out_right
            }
        }

        viewBinding.registerButton.setOnClickListener(
            Navigation.createNavigateOnClickListener(R.id.next_action, null)
        )


//        viewBinding.alreadyHaveAnAccount.setOnClickListener {
//            findNavController().navigate(R.id.loginFragment, null, options)
//        }
        return viewBinding.root
    }

}